### PR TITLE
Add more check on softmax ONNX exporting logic

### PR DESCRIPTION
**Pull request by houseroad**
PyTorch: split the content based on the dim
ONNX: squeeze the tensor into 2d, and do the softmax on it.

So their semantics are different.

### CODE REVIEW

**Comment by houseroad** [comment]
The semantics match if the dim equals ndim-1, which means it sum the results along the last dimension.

**Comment by ezyang** [comment]
It would be nice if there was a little commentary in the if-test about why it is hard to do properly (as we discussed offline.)

**Comment by houseroad** [comment]
More comments are added to explain the differences between axis and dim.
